src/prommis/solvent_extraction/tests/test_sx_model_dynamic.py

```diff
@@ -0,0 +1,295 @@
+    m.fs.solex.mscontactor.aqueous[0, :].conc_mass_comp["La"].fix(1e-7)
+    m.fs.solex.mscontactor.aqueous[0, :].conc_mass_comp["Ce"].fix(1e-7)
+    m.fs.solex.mscontactor.aqueous[0, :].conc_mass_comp["Pr"].fix(1e-7)
+    m.fs.solex.mscontactor.aqueous[0, :].conc_mass_comp["Nd"].fix(1e-7)
+    m.fs.solex.mscontactor.aqueous[0, :].conc_mass_comp["Sm"].fix(1e-7)
+    m.fs.solex.mscontactor.aqueous[0, :].conc_mass_comp["Gd"].fix(1e-7)
+    m.fs.solex.mscontactor.aqueous[0, :].conc_mass_comp["Dy"].fix(1e-7)
+
+    m.fs.solex.mscontactor.aqueous[0, :].conc_mass_comp["SO4"].fix(3999.818)
+    m.fs.solex.mscontactor.aqueous[0, :].conc_mass_comp["HSO4"].fix(693.459)
+    m.fs.solex.mscontactor.aqueous[0, :].conc_mass_comp["Al"].fix(422.375)
+    m.fs.solex.mscontactor.aqueous[0, :].conc_mass_comp["Ca"].fix(109.542)
+    m.fs.solex.mscontactor.aqueous[0, :].conc_mass_comp["Cl"].fix(1e-7)
+    m.fs.solex.mscontactor.aqueous[0, :].conc_mass_comp["Fe"].fix(688.266)
+    m.fs.solex.mscontactor.aqueous[0, :].conc_mass_comp["Sc"].fix(0.032)
+    m.fs.solex.mscontactor.aqueous[0, :].conc_mass_comp["Y"].fix(0.124)
+    m.fs.solex.mscontactor.aqueous[0, :].conc_mass_comp["La"].fix(0.986)
+    m.fs.solex.mscontactor.aqueous[0, :].conc_mass_comp["Ce"].fix(2.277)
+    m.fs.solex.mscontactor.aqueous[0, :].conc_mass_comp["Pr"].fix(0.303)
+    m.fs.solex.mscontactor.aqueous[0, :].conc_mass_comp["Nd"].fix(0.946)
+    m.fs.solex.mscontactor.aqueous[0, :].conc_mass_comp["Sm"].fix(0.097)
+    m.fs.solex.mscontactor.aqueous[0, :].conc_mass_comp["Gd"].fix(0.2584)
+    m.fs.solex.mscontactor.aqueous[0, :].conc_mass_comp["Dy"].fix(0.047)
+
+    m.fs.solex.mscontactor.aqueous_inherent_reaction_extent[0, :, "Ka2"].fix(0)
+    m.fs.solex.mscontactor.aqueous[0, :].flow_vol.fix(62.01)
+
+    m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Al"].fix(1e-7)
+    m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Ca"].fix(1e-7)
+    m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Fe"].fix(1e-7)
+    m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Sc"].fix(1e-7)
+    m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Y"].fix(1e-7)
+    m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["La"].fix(1e-7)
+    m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Ce"].fix(1e-7)
+    m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Pr"].fix(1e-7)
+    m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Nd"].fix(1e-7)
+    m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Sm"].fix(1e-7)
+    m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Gd"].fix(1e-7)
+    m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Dy"].fix(1e-7)
+
+    m.fs.solex.mscontactor.organic[0, :].flow_vol.fix(62.01)
+
+    return m
+
+
+@pytest.mark.unit
+def test_structural_issues(model):
+    dt = DiagnosticsToolbox(model)
+    dt.report_structural_issues()
+    dt.assert_no_structural_warnings(ignore_unit_consistency=True)
```

Comment:
Why is the unit consistency being ignored? Is this just an ongoing issue with all dynamic flowsheets?

Reply:
There are derivative terms being included here, so during discretization, the unit of the denominator goes away. This issue is with any derivative term, as far as I know.